pkg/operator/tektonchains.go

```diff
@@ -0,0 +1,83 @@
+package operator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/openshift-pipelines/release-tests/pkg/assert"
+	"github.com/openshift-pipelines/release-tests/pkg/clients"
+	"github.com/openshift-pipelines/release-tests/pkg/config"
+	"knative.dev/pkg/test/logging"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	operatorv1alpha1 "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EnsureTektonChainExists creates a TektonChain with the name names.TektonChain, if it does not exist.
+func EnsureTektonChainExists(clients operatorv1alpha1.TektonChainInterface, names config.ResourceNames) (*v1alpha1.TektonChain, error) {
+	// If this function is called by the upgrade tests, we only create the custom resource, if it does not exist.
```

Comment:
Hi @ppitonak , where you defined this struct config.ResourceNames ?

Reply:
it's still WIP, it should have been probably `utils.ResourceNames` from github.com/tektoncd/operator/test/utils